### ISSUE TITLE
Model checkpointing

### ISSUE THREAD

**Issue by mmatena** [mmatena/del8#9]
I think this can actually be done fairly easily with my existing training framework. The steps would be like:

- See if a CheckpointSummary already exists for a model with the RunKey.
- If not, then train as we are now. Otherwise:
- Load the latest checkpoint from the summary, the summary itself, and adjust some parameters of training to reflect that we are not starting from the beginning.

Some things to consider:
- Right now we are doing `model.save_weights` to save checkpoints, which IIRC might not be the best for this sort of model checkpointing and restarting.
- Maybe related to the previous point, there are other variables such as train step and optimizer internal variables that need to be correct updated. I need to make sure that things like learning rate schedules that depend on these are written such that they can be reloaded to restart training from a checkpoint.
- We'd need some failure handling and auto-restart (and probably some code to make sure we don't restart endlessly if there is, for example, a code bug that causes training to always fail) on the supervisor.
